packages/oauth/src/install-provider.ts

```diff
@@ -336,48 +449,65 @@ export class InstallProvider {
   ): Promise<void> {
     let code: string;
     let flowError: string;
-    let state: string;
+    let stateInQueryString: string;
     try {
       if (req.url !== undefined) {
         // Note: Protocol/ host of object are not necessarily accurate
         // and shouldn't be relied on
         // intended only for accessing searchParams only
-        const searchParams = InstallProvider.extractSearchParams(req);
+        const searchParams = extractSearchParams(req);
         flowError = searchParams.get('error') as string;
         if (flowError === 'access_denied') {
           throw new AuthorizationError('User cancelled the OAuth installation flow!');
         }
         code = searchParams.get('code') as string;
-        state = searchParams.get('state') as string;
+        stateInQueryString = searchParams.get('state') as string;
         if (!code) {
           throw new MissingCodeError('Redirect url is missing the required code query parameter');
         }
-        if (this.stateVerification && !state) {
+        if (this.stateVerification && !stateInQueryString) {
           throw new MissingStateError('Redirect url is missing the state query parameter. If this is intentional, see options for disabling default state verification.');
         }
       } else {
         throw new UnknownError('Something went wrong');
       }
       // If state verification is enabled, attempt to verify, otherwise ignore
-      if (this.stateVerification && this.stateStore) {
-        // eslint-disable-next-line no-param-reassign
-        installOptions = await this.stateStore.verifyStateParam(new Date(), state);
+      if (this.stateVerification) {
+        try {
+          if (this.legacyStateVerification) {
+            // This mode is not enabled by default
+            // This option is for some of the existing developers that need time for migration
+            this.logger.warn('Enabling legacyStateVerification is not recommended as it does not properly work for OAuth CSRF protection. Please consider migrating from directly using InstallProvider#generateInstallUrl() to InstallProvider#handleInstallPath() for serving the install path.');
+          } else {
+            const stateInBrowserSession: string | undefined = extractCookieValue(req, this.stateCookieName);
+            if (!stateInBrowserSession || (stateInBrowserSession !== stateInQueryString)) {
```

Comment:
In my use case the state is stored in a database. I created a custom State Store.

The client is a React app, and the backend is "AWS AppSync to Lambda" configuration, hence, not a nodejs web server.
I do not have nodejs `request` and `response` objects, nor access to the browser's cookies.

I am using `node-slack-sdk` to generate the install URL and to handle the callback, however, handleCallback is executed from a lambda function, without access to a browser.

`stateInBrowserSession` is always undefined in my case, and fails this check.

I am setting `legacyStateVerification` to true to avoid this.
I don't think the warning message is required in this case, and I ask you to keep `legacyStateVerification` since it is not only for `for backward-compatibility with v2.4 and olders` but also for backends that use custom state stores and do not have access to browser cookies